components/Heading/__tests__/Heading.test.js

```diff
@@ -16,16 +20,248 @@ describe('Heading', () => {
       </Heading>,
     );
   });
+});
+
+describe('Heading anchor and hashlink', () => {
```

Comment:
I appreciate that you went after total coverage, but keep in mind we have to maintain tests and we want to write them to instill confidence in our code (by protecting against regressions and asserting complex things work as we expect).

Keeping that in mind, we can drastically reduce the amount of tests here.

I'd say have:
- One test that the component renders at all with required props (snapshot test is fine here)
- Two tests that assert the link's [visibility/invisibility](https://github.com/testing-library/jest-dom#tobevisible)

and that's it. You don't want to get caught testing against class names or asserting that props get passed through. You want to think about it from the perspective of the user interacting with it. They just care that the heading component works (renders) and functions properly.

Testing it's functionality is more valuable in a browser-like env, like Cypress because you can assert that the scroll position gets changed and the URL gets changed. The only functionality you care about in Jest is the anchor being visible after `onMouseEnter` or when focused **AND** invisible after `onMouseLeave` and when not focused.

Reply:
I ended up removing all the tests except for the Cypress ones. The icon is based on hover now and there is no more state to check visibility.